### PR TITLE
Add xUnit UWP tools to buildtools.

### PR DESCRIPTION
**Pull request by crummel**
These changes allow running NetCoreForCoreCLR tests in an AppX container
using Helix.
- Adds a console tool for installing and setting up the AppX.
- Adds an AppX that actually runs the xUnit tests and writes out the results.

### CODE REVIEW

**Comment by crummel** [comment]
Ping @karajas, @Chrisboh, @weshaggard, @roncain

**Comment by Chrisboh** [comment]
@MattGal

File: build.proj [hunk 1]
```diff
@@ -42,9 +42,9 @@
 
   <Target Name="ValidateAllProjectDependencies"
           Condition="'$(ValidatePackageVersions)'=='true' and '@(ProjectJsonFiles)'!=''">
-    <ValidateProjectDependencyVersions ProjectJsons="@(ProjectJsonFiles)"
+<!--    <ValidateProjectDependencyVersions ProjectJsons="@(ProjectJsonFiles)"
```
Comment (MattGal) [inline build.proj]: I find it generally useful to put a comment over commented-out sections explaining why.

File: src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/xunitrunner-appx/xunitrunner-appx.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env py
```
Comment (Chrisboh) [inline src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/xunitrunner-appx/xunitrunner-appx.py]: We shouldn't need this runner script anymore correct?  My understanding is all of this logic belongs with the product cmd file.

Comment (weshaggard) [inline src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/xunitrunner-appx/xunitrunner-appx.py]: I certainly hope we don't need this :)

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets [hunk 3]
```diff
@@ -28,25 +28,29 @@
...
 
-  <ItemGroup Condition="'$(TestWithCore)' == 'true'">
+  <ItemGroup Condition="'$(TestWithCore)' == 'true' And '$(TestAppX)' != 'true'">
     <TestTargetFramework Include="DNXCore,Version=v5.0">
       <Folder>dnxcore50</Folder>
     </TestTargetFramework>
   </ItemGroup>
 
+  <PropertyGroup Condition="'$(TestAppX)' == 'true'">
+    <XunitExecutable Condition="'$(XunitExecutable)' == ''">%HELIX_CORRELATION_PAYLOAD%\RunnerScripts\xunitrunner-appx\xunitrunner-appx.py</XunitExecutable>
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets]: :( I don't want us to start depending on python for our tests, and I also want these to be able to run locally.

**Comment by crummel** [comment]
Changed this to just be the tooling, build changes will come soon in a separate pull request.

File: src/xunit.console.uwp/CommandLine.cs [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+                throw new ArgumentException($"error: unknown command line option: {option.Value}");
+        }
+
+        static bool IsConfigFile(string fileName)
+        {
+            return fileName.EndsWith(".config", StringComparison.OrdinalIgnoreCase)
+                || fileName.EndsWith(".json", StringComparison.OrdinalIgnoreCase);
+        }
+
+        public static CommandLine Parse(params string[] args)
```
Comment (Chrisboh) [inline src/xunit.console.uwp/CommandLine.cs]: Most of this code is duplicated between here and the console app.  Could we unify this code and share it between the two?

**Comment by crummel** [comment]
I think all the comments so far were addressed - thanks @karajas, @Chrisboh, @weshaggard, @roncain

**Comment by Chrisboh** [comment]
:shipit:

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets [hunk 5]
```diff
@@ -28,15 +28,15 @@
     <TestWithCore Condition="'$(TestWithCore)' == ''">true</TestWithCore>
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(TestWithCore)' == 'true'">
+  <PropertyGroup Condition="'$(TestWithCore)' == 'true' And '$(TestAppX)' != 'true'">
```
Comment (karajas) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets]: So currently is it correct to assume that when we test UWP, we will run build.cmd /p:TestTFM=netcore50 /p:TestAppX=true ?

**Comment by weshaggard** [comment]
I didn't look deeply through the code (don't block on me for that) but the general hook-up of just switching the host in this mode looks along the lines of what I wanted to see. Thanks.

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets (hunk 5 above)
Comment (crummel) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets]: Yes, that's correct.

**Pull request merged**